pandas/io/parsers.py

```diff
@@ -2442,6 +2450,14 @@ def _handle_usecols(self, columns, usecols_key):
                     raise ValueError("If using multiple headers, usecols must "
                                      "be integers.")
                 col_indices = []
+
+                missing = [c for c in self.usecols if c not in usecols_key]
```

Comment:
I'm unsure what design patterns Pandas follows for this kind of thing, but would you rather me drop this down into a try/catch for `col_indices.append(usecols_key.index(col))` where the error is currently being thrown?

I worry this initial approach adds needless computation time - but unsure if it's more readable? Let me know 😄

Reply:
Could you elaborate further on that logic you provided: `col_indices.append(usecols_key.index(col))`.  I'm not sure I fully follow where / how this would be added.

Secondly, don't worry about computation time.  Get a working implementation first.  Then we'll worry about optimizing, if need be.  Chances are that won't be an issue.

Reply:
Sure, at the moment the error is being raised a few lines further down when we're looking for the index of the column in the `usecols_key` list.

The alternate proposal would be something like:
```
try:
    col_indices.append(usecols_key.index(col))
except ValueError:
    # Calculate missing usecols and raise error accordingly
```

Reply:
Ah, understood.  Yes, absolutely, let's do `try-except`.